### PR TITLE
[Security Content] Add Windows Investigation Guides

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Related to https://github.com/elastic/security-team/issues/7664

## Summary

Adds Investigation Guides to the following rules:

* Potential File Transfer via Certreq
* Ingress Transfer via Windows BITS
* NTDS or SAM Database File Copied
* LSASS Process Access via Windows API
* Network Logon Provider Registry Modification
* PowerShell Kerberos Ticket Dump
* Potential Credential Access via Renamed COM+ Services DLL
* Group Policy Discovery via Microsoft GPResult Utility
* Potential PowerShell HackTool Script by Function Names
* Persistence via PowerShell profile

### CODE REVIEW

File: rules/windows/command_and_control_certreq_postdata.toml [hunk 1]
```diff
@@ -16,6 +41,59 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
 language = "eql"
 license = "Elastic License v2"
 name = "Potential File Transfer via Certreq"
+note = """## Triage and analysis
+
+### Investigating Potential File Transfer via Certreq
+
+Certreq is a command-line utility in Windows operating systems that allows users to request and manage certificates from certificate authorities. It is primarily used for generating certificate signing requests (CSRs) and installing certificates. However, adversaries may abuse Certreq's functionality to download files or upload data to a remote URL by making an HTTP Post request.
```
Comment (joepeeples) [inline rules/windows/command_and_control_certreq_postdata.toml:48]:
```suggestion
Certreq is a command-line utility in Windows operating systems that allows users to request and manage certificates from certificate authorities. It is primarily used for generating certificate signing requests (CSRs) and installing certificates. However, adversaries may abuse Certreq's functionality to download files or upload data to a remote URL by making an HTTP POST request.
```

File: rules/windows/command_and_control_certreq_postdata.toml [hunk 2]
```diff
@@ -16,6 +41,59 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*", "endgame-*"
...
+- Contact the account owner and confirm whether they are aware of this activity.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Assess whether this behavior is prevalent in the environment by looking for similar occurrences across hosts.
+- Examine the details of the dropped file, and whether it was executed.
+- Check the reputation of the domain or IP address used to host the downloaded file or if the user downloaded the file from an internal system.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the file using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/command_and_control_certreq_postdata.toml:68]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml [hunk 3]
```diff
@@ -18,6 +43,62 @@ language = "eql"
...
+- Contact the account owner and confirm whether they are aware of this activity.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Check for any recent changes in user account privileges or group memberships that may have allowed the unauthorized access.
+- Determine whether the file was potentially exfiltrated from the subject host.
+- Scope compromised credentials and disable the accounts.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the process executable using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml:70]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/credential_access_lsass_openprocess_api.toml [hunk 4]
```diff
@@ -16,6 +41,70 @@ index = ["logs-endpoint.events.*"]
...
+  - Investigate any abnormal behavior by the subject process, such as network connections, DLLs loaded, registry or file modifications, and any spawned child processes.
+- Assess the access rights (`process.Ext.api.parameters.desired_access`field) requested by the process. This [Microsoft documentation link](https://learn.microsoft.com/en-us/windows/win32/procthread/process-security-and-access-rights) may be useful to help the interpretation.
+- If there are traces of LSASS memory being successfully dumped, investigate potentially compromised accounts. Analysts can do this by searching for login events (e.g., 4624) to the target host.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the executables of the processes using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/credential_access_lsass_openprocess_api.toml:70]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/credential_access_persistence_network_logon_provider_modification.toml [hunk 5]
```diff
@@ -19,6 +44,61 @@ index = ["logs-endpoint.events.*", "endgame-*"]
...
+  - Validate the activity is not related to planned patches, updates, network administrator activity, or legitimate software installations.
+  - Investigate any abnormal behavior by the subject process, such as network connections, DLLs loaded, registry or file modifications, and any spawned child processes.
+- Retrieve the file and examine if it is signed with valid digital signatures from vendors that are supposed to implement this kind of software and homologated to use in the environment. Check for prevalence in the environment and whether they are located in expected locations.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the executables of the processes using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/credential_access_persistence_network_logon_provider_modification.toml:70]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/credential_access_suspicious_comsvcs_imageload.toml [hunk 6]
```diff
@@ -18,6 +43,68 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+
+- Investigate the process execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Investigate any abnormal behavior by the subject process, such as network connections, registry or file modifications, and any spawned child processes.
+- Identify the process that created the DLL using file creation events.
+   - Inspect the file for useful metadata, such as file size and creation or modification time.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the process executable and DLL using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/credential_access_suspicious_comsvcs_imageload.toml:67]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

File: rules/windows/execution_posh_hacktool_functions.toml [hunk 7]
```diff
@@ -17,6 +42,71 @@ index = ["winlogbeat-*", "logs-windows.*"]
...
+- Investigate the script execution chain (parent process tree) for unknown processes. Examine their executable files for prevalence, whether they are located in expected locations, and if they are signed with valid digital signatures.
+- Examine the script's execution context, such as the user account, privileges, the role of the system on which it was executed, and any relevant timestamps.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Evaluate whether the user needs to use PowerShell to complete tasks.
+- Investigate the origin of the PowerShell script, including its source, download method, and any associated URLs or IP addresses.
+- Examine the host for derived artifacts that indicate suspicious activities:
+  - Analyze the script using a private sandboxed analysis system.
+  - Observe and collect information about the following activities in both the sandbox and the alert subject host:
+    - Attempts to contact external domains and addresses.
+      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process' `process.entity_id`.
```
Comment (joepeeples) [inline rules/windows/execution_posh_hacktool_functions.toml:68]:
```suggestion
      - Use the Elastic Defend network events to determine domains and addresses contacted by the subject process by filtering by the process's `process.entity_id`.
```

**Review by joepeeples — COMMENTED** [review]
Added some small line-edits. Thanks for preparing this content!

**Review by terrancedejesus — APPROVED** [review]
Great work!

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**